Sources/SwiftDocC/Infrastructure/External Data/OutOfProcessReferenceResolver.swift

```diff
@@ -257,6 +261,50 @@ public class OutOfProcessReferenceResolver: ExternalReferenceResolver, FallbackR
             semantic: symbol,
             platformNames: resolvedInformation.platformNames
         )
+        
+        addImagesAndCacheMediaReferences(to: &node, from: resolvedInformation)
+        
+        return node
+    }
+    
+    private func addImagesAndCacheMediaReferences(to node: inout DocumentationNode, from resolvedInformation: ResolvedInformation) {
+        // Because DocC renders external content in the local context, external media also needs to be added to the local context.
+        // If the external content was treated as pre-rendered then this wouldn't be necessary. (rdar://78718811)
+        // FIXME: https://github.com/apple/swift-docc/issues/468
+        
+        // `@PageImage` directives isn't meant to be created in code like this but it's the only way to associate topic images
+        // with a render node.
+        
+        if let topicImages = resolvedInformation.topicImages, !topicImages.isEmpty {
+            let metadata = node.metadata ?? Metadata(originalMarkup: BlockDirective(name: "Metadata", children: []), documentationExtension: nil, technologyRoot: nil, displayName: nil)
+            
+            metadata.pageImages = topicImages.map { topicImage in
+                let purpose: PageImage.Purpose
+                switch topicImage.type {
+                case .card: purpose = .card
+                case .icon: purpose = .icon
+                }
+                
+                return PageImage._make(
+                    originalMarkup: BlockDirective(name: "PageImage", children: []),
+                    purpose: purpose,
+                    source: ResourceReference(bundleIdentifier: node.reference.bundleIdentifier, path: topicImage.identifier.identifier),
+                    alt: (resolvedInformation.references?.first(where: { $0.identifier == topicImage.identifier }) as? ImageReference)?.altText
```

Comment:
Do we need this? I _think_ it should pull the alt text from the reference at render time.

Reply:
Yeah, unfortunately we can't return the image reference from the external resolver (there are only return values for the `DocumentationNode` and a separate function that returns the `DataAsset`) so we have to break apart the image reference so that DocC can reassemble it into a new image reference with the same data.